src/tcp-intercept.cxx

```diff
@@ -473,6 +505,9 @@ int main(int argc, char* argv[]) {
 #if HAVE_DECL_IP_TRANSPARENT
 		int value = 1;
 		s_listen.setsockopt(SOL_IP, IP_TRANSPARENT, &value, sizeof(value));
+		s_listen.setsockopt(IPPROTO_TCP, TCP_NODELAY, (char *) &value, sizeof(value));
+		//Take care of socks hanging in Established state
+		s_listen.setsockopt(SOL_SOCKET,SO_KEEPALIVE, &value, sizeof(value));
```

Comment:
- Activate/deactivate TCP_NODELAY:
  This could be done via sysctl interface:
       net.ipv4.tcp_low_latency = 0/1
- Setting the keepalive option on the listening socket:
That was not my original idea either. After a day of unsuccessful attempts to set it only on the s_server socket, I tried (just out of curiosity) to set it on the listening socket and to my surprise it worked. I am still puzzled as to why!  I just made some tests and seems setting the option in the s_socket is redundant and could be removed.